test/command_line/test_integrate.py

```diff
@@ -111,6 +112,27 @@ def test_basic_integrate(dials_data, tmpdir):
     # assert(flex.abs(diff_Obs_P).all_lt(1e-7))
 
 
+@pytest.mark.parametrize(
+    ("block_size", "block_units"),
+    [(None, None), (1, "degrees"), (2, "frames"), (1, "frames")],
+)
+def test_basic_blocking_options(dials_data, tmpdir, block_size, block_units):
+    exp = load.experiment_list(
+        dials_data("centroid_test_data").join("experiments.json").strpath
+    )
+    exp[0].identifier = "foo"
+    exp.as_json(tmpdir.join("modified_input.json").strpath)
+
+    args = ["dials.integrate", "modified_input.json", "nproc=2"]
+    if block_size:
+        args.append("block.size=%s" % block_size)
+    if block_units:
+        args.append("block.units=%s" % block_units)
+
+    result = procrunner.run(args, working_directory=tmpdir)
+    assert not result.returncode and not result.stderr
```

Comment:
```suggestion
def test_basic_blocking_options(dials_data, tmp_path, block_size, block_units):
    exp = load.experiment_list(
        dials_data("centroid_test_data") / "experiments.json"
    )
    exp[0].identifier = "foo"
    exp.as_json(tmp_path / "modified_input.json")

    args = ["dials.integrate", "modified_input.json", "nproc=2"]
    if block_size:
        args.append("block.size=%s" % block_size)
    if block_units:
        args.append("block.units=%s" % block_units)

    result = procrunner.run(args, working_directory=tmp_path)
    assert not result.returncode and not result.stderr
```
* `load.experiment_list` [explicitly can handle path objects](https://github.com/cctbx/dxtbx/blob/d17b7ccb1bcb340ef9e121c27bb4367930fbb794/serialize/load.py#L119-L122)
* `.as_json()` _should_ be able to handle path objects
* prefer `tmp_path` over `tmpdir` (Python [standard library `pathlib` object](https://docs.python.org/3/library/pathlib.html) rather than [custom & deprecated `py.path` object](https://py.readthedocs.io/en/latest/path.html))
* pathlib objects don't support `.join` though, so need to replace that with either [`/`](https://docs.python.org/3/library/pathlib.html#basic-use) or [`.joinpath`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.joinpath)

This is untested 🙂